## Who are you? A framework to identify and report genetic sample mix-ups

### Synopsys
We surveyed the community and found that almost 80% of responding researchers have encountered sample mix-ups. However, many recent studies in the field of molecular ecology do not appear to systematically report individual assignment checks as part of their publications. Although checks may be done, lack of consistent reporting means that it is difficult to assess whether sample mix-ups have occurred or been detected. Here, we present an easy-to-follow sample verification framework that can utilise existing metadata, including species, population structure, sex and pedigree information.

#### Detection of sample mix-ups
Initial comparison of heterozygosity values of genotyped Z-linked SNP array markers for the Tiritiri Matangi individuals. Later, sample mix-up was confirmed comparing pedigree (microsatellite-based) and genomic relatedness in R.

#### Blank, Control and Duplicate check
Duplicates were identified by the default criteria in the Axiom Analysis Suite software. Sample duplicate individuals can also be identified for example using the --genome function in the PLINK software, in COLONY during analysis set-up or from the sequoia function in the package sequoia in R.

#### Principal Component Analysis 
We used the PLINK function –make-rel to calculate relatedness and the function --pca to get principal components. We used R to plot the first two principal components and the third and fourth principal components.

#### Sex check
Z-linked SNPs were used to confirm the sex of the genotyped individuals by calculating homozygosity by locus for 2,694 sex markers in the 'GENHET' package in R. For comparison, we also made use of the –check-sex function in PLINK, with the default 0.2/0.8 F-statistic threshold.

#### Check for relatedness consistency 
As a next step, we compared the pedigree-based (kinship2) and genomic relatedness (GCTA) of first-degree relationships of the individual hihi in order to identify more sample errors. As PLINK only reports Mendelian errors per family, sequoia only per SNP, and COLONY only indirectly, we instead built a custom-matrix in R. 

#### Parentage assignment 
We used the R package sequoia and the stand-alone software COLONY to reconstruct pedigree relationships for all Tiritiri Matangi individuals. Sequoia makes use of a conservative hill-climbing algorithm to construct high likelihood pedigrees using relatively few markers, however, this heuristic, sequential approach also means that the inferred relationship between two individuals is not necessarily the true relationship (Huisman, 2017). COLONY on the other hand can employ one likelihood or two pairwise likelihood methods for inferring parentage and sibship (Jones & Wang, 2010; Wang, 2013), and has the advantage of being rather conservative and hence producing trustworthy pedigrees where the assignment confidences are reported separately.
The COLONY windows GUI helped create the input .dat file for COLONY for 1,079 of the hihi which have at least one genotyped parent. Data input comprised 939 SNPs in ‘1234 format’ after filtering in PLINK (--maf 0.4 --indep-pairwise 50 5 0.05 on all 45,553 SNPs; plus option --recode 12) and included 126 male and 110 female candidates that are offspring as well as potential parents to some of the 1079 offspring. The seed for the random number generator was set to 1234, a genotyping error rate of 0.01 and the number of known paternal or maternal sibships set to zero. The probability of the true parents being in the candidate lists was conservatively set at 0.5 for females and 0.5 for males to reflect the very high probability of recapture of the individuals on the island (Chauvenet et al., 2013). The inbreeding model was selected, polygamous mating system for both males and females allowed and the program ran using the FL-PLS-combined (FPLS) method at medium length.
For the analysis in sequoia, the COLONY files were transformed to .RAW (0/1/2) format in PLINK, however, sequoia has recently been updated to be able to read in many different data formats. An additional life history file with the sex and year of birth of the individuals was provided for the sequoia function performing sibship clustering and grandparent assignment. All 1,250 individuals were included in this analysis but the variants used previously in COLONY were further subsampled to 400 random SNPs in R as recommended by the author (Huisman, 2017). Further options for the full pedigree reconstruction in sequoia included the allowance for a genotyping error rate of 0.01, three mismatches per pair and a maximum number of eight iterations. All other options were set as default, as per the sequoia mid-2019 version.
For both sequoia and COLONY, dummy parents are assigned if other genotyped individuals cannot be assigned as parents. Parentage assignment in sequoia inferred a multigenerational pedigree where the 1,250 genotyped hihi have either a real hihi or
one (or two) of 133 dummy individuals assigned as parents. In COLONY, a more conservative tool, the new genotype-based pedigree included 119 dummy sires to 545 hihi and 159 dummy dams to 550 hihi. When comparing the verified pedigree to these two family reconstructions, for 354 samples the same two parents were assigned in all three cases. This indicates that these samples have correct ID-genotype associations all throughout the data processing and are safe to work with in future projects. In 107 cases, no pedigree agreed with each other. In another 42 cases, the two softwares agreed on a parental pair for a sample but are different from those noted in the verified pedigree. Those genotypes can be marked as ‘rejected’, but might also be resolved by backtracking which individual has those two parents assigned in the verified pedigree. This procedure is only possible for only-child offspring, saving 20 of the 42 individuals.
Further scrutinizing of the data might resolve the identity of some more mix-ups, but we want to acknowledge that hundreds of the samples may never be recovered unless we perform further genotyping from the original sample tubes, which we are confident are not mixed-up. What is more, we have only included individuals for the pedigree reconstruction that were genotyped from plate wells assigned to a Tiritiri Matangi bird; hence, more birds could be recovered if all array genotypes are considered simultaneously. Individuals whose field and genetic sex matched but without any first-degree relatives were conservatively considered unvalidated. Plotting of the relatedness values for confirmed and rejected first-degree Tiritiri Matangi relationships (Figure S3) support our claim that none of the rejected samples are indeed correct. On the other hand, the confirmed cases only show few outliers, which can be due to variation in inbreeding, and mostly show that the pedigree relatedness of those pairs is similar to the one observed when comparing their genotypes.

#### The use of whole-genome sequencing data
Another way of identifying sample errors is to calculate identity scores for genomic input of different resolutions for the same individuals, as confirmed by the answers of the 2020 survey. This means that, for instance, previously generated whole-genome sequencing data can help confirm that newly generated SNP datasets for the same individuals are correctly assigned the same ID. For ten hihi, we had whole-genome sequencing (WGS) data available that covered almost four million SNPs in their genome. In order to check that these WGS individuals were correctly allocated on the SNP array, we reduced the WGS files to those sites that are also genotyped with the hihi SNP array. Those files from two different genotyping processes were merged, resulting in a file containing each of the ten individuals twice. PLINK’s --genome function then confirmed that all ten hihi pairs had a pairwise IBD estimate of one, meaning that those ten hihi were assigned the correct ID during the SNP array genotyping. In this case, the three Tiritiri Matangi birds had been previously confirmed from the SNP array data only (correct sex, correct pedigree links) but we were able to verify that the seven WGS individuals from Te Hauturu-o-Toi were also correct.

#### Identifying sources of error 
For the Tiritiri Matangi population of hihi, a summary of the confirmed, corrected, unvalidated and rejected ID-genotype associations can be found in Table 2 in the main manuscript. In order to address whether wet lab errors arose from systematic mis-plating of samples, we mapped confirmed, unvalidated and rejected individuals across all genotyping plates. For this, we made use of the R package platetools (https://cran.r-project.org/package=platetools) that allowed us to colour code the wells according to the sample classification. When checking the confirmed versus rejected samples, we could not identify any systematic pattern of how the samples had been mixed-up. All plates seem to have been affected by sample mix-ups, with some plates having their samples more easily confirmed than others (Figure S4). There was no clear pattern with regard to only specific cohorts being affected by the mix-up, as affected samples spanned all years. All plates had at least one confirmed individual, which suggests that none of the plates were accidentally rotated by 180 degrees, which was an error mentioned by three participants in the online survey we conducted (see above).
Overall, while we were not able to trace back precisely the origin of the sample mix-ups of the hihi genotypes, we suspect that most errors occurred in the laboratory.
