
# Genomic insights into the adaptive potential of the hihi (*Notiomystis cincta*), a threatened Aotearoa New Zealand bird

Presented here are a list of tools and example scripts used in each data chapter of my thesis. Within each directory is a brief overview of chapter contents and details about each of the tools used. In recognition of Indigenous Data sovereignty of data related to culturally significant species, any sample IDs have been redacted from these scripts and data files are only available on request.


The thesis is devided into four data chapters:

## Chapter 2: A framework to resolve sample mix-ups
We surveyed the community and found that almost 80% of responding researchers have encountered sample mix-ups. However, many recent studies in the field of molecular ecology do not appear to systematically report individual assignment checks as part of their publications. Although checks may be done, lack of consistent reporting means that it is difficult to assess whether sample mix-ups have occurred or been detected. Here, we present an easy-to-follow sample verification framework that can utilise existing metadata, including species, population structure, sex and pedigree information.

## Chapter 3: A GWAS analysis for hihi adaptive traits
We estimated the heritability of three hihi (*Notiomystis cincta*) morphological traits known to be under selection (nestling tarsus length, body mass and head‚Äìbill length) using 523 individuals and 39 699 single nucleotide polymorphisms (SNPs) from a 50 K Affymetrix SNP chip. We then examined the genetic architecture of the traits via chromosome partitioning analyses and genome-wide association scans (GWAS).

## Chapter 4: A comparison of different data resolutions to estimate inbreeding
We performed an in-depth inbreeding analysis for hihi (*Notiomystis cincta*), an endemic and nationally vulnerable passerine bird of Aotearoa New Zealand. We first focused on subsampling variants from a reference genome male and then extended our analysis to 10 hihi where low-coverage whole genome resequencing, RAD-seq and SNP array genotypes are available. 

## Chapter 5: An inbreeding and inbreeding depression analysis
Here, we investigate inbreeding and inbreeding depression for hihi (*Notiomystis cincta*). Firstly, using a custom 45k SNP array, we explore genomic inbreeding patterns by inferring homozygous segments across the genome. Secondly, we measure genomic inbreeding effects on fitness, measured as lifetime reproductive success, and its three components, annual reproductive success, juvenile survival and adult annual survival, in 363 hihi. 


### Publications
Publications associated with this thesis include: 

#### [Published Article](https://doi.org/10.1098/rspb.2020.0948) in Proceedings of the Royal Society B
#### [Published Article](https://doi.org/10.1111/mec.16068) in the Molecular Ecology Special Issue: Whole-genome sequencing in Molecular Ecology
#### [Published Article](https://doi.org/10.1111/1755-0998.13575) in Molecular Ecology Resources

Please don't hesitate to contact me with questions regarding any of the chapters üê¶ üí¨
