
# Genomic insights into the adaptive potential of the hihi (*Notiomystis cincta*), a threatened Aotearoa New Zealand bird

Presented here are a list of tools and example scripts used in each data chapter of my thesis. Within each directory is a brief overview of chapter contents and details about each of the tools used. In recognition of Indigenous Data sovereignty of data related to culturally significant species, any sample IDs have been redacted from these scripts and data files are only available on request.


The thesis is devided into four data chapters:

## Chapter 2: A framework to resolve sample mix-ups
We then examined the genetic architecture of the traits via chromosome partitioning analyses and genome-wide association scans (GWAS). Heritabilities estimated using pedigree relatedness or genomic relatedness were low. For tarsus length, the proportion of genetic variance explained by each chromosome was positively correlated with its size, and more than one chromosome explained significant variation for body mass and head‚Äìbill length. Finally, GWAS analyses suggested many loci of small effect contributing to trait variation for all three traits, although one locus (an SNP within an intron of the transcription factor HEY2) was tentatively associated with tarsus length.

## Chapter 3: A GWAS analysis for hihi adaptive traits


## Chapter 4: A comparison of different data resolutions to estimate inbreeding


## Chapter 5: An inbreeding and inbreeding depression analysis



Don't hesitate to contact me with questions regarding any of the procedures üê¶ üí¨

### Publications
Publications associated with this thesis include: 

i) Published Article in Proceedings of the Royal Society B
ii) Published Article in the Molecular Ecology Special Issue: Whole-genome sequencing in Molecular Ecology
iii) Published Article in Molecular Ecology Resources
