
## Polygenic basis for adaptive morphological variation in a threatened Aotearoa | New Zealand bird, the hihi (*Notiomystis cincta*)

We estimated the heritability of three hihi (*Notiomystis cincta*) morphological traits known to be under selection (nestling tarsus length, body mass and head–bill length) using 523 individuals and 39 699 single nucleotide polymorphisms (SNPs) from a 50 K Affymetrix SNP chip. We then examined the genetic architecture of the traits via chromosome partitioning analyses and genome-wide association scans (GWAS). Detailed methods can be found in the [publication](https://doi.org/10.1098/rspb.2020.0948).

#### Genotyping and pedigree data
Of the 58 466 SNPs included on a costum array, 45 553 markers were designated as polymorphic high resolution according to the default quality control metrics in the Axiom Analysis Suite software. PLINK filtering (--maf 0.01, --hwe 1e-20) and duplicate removal followed (i.e. mapped to the same chromosome at the same position). No individual was removed due to low genotyping rates (--mind).

#### Trait heritabilities
Pedigree- and GRM-based heritabilities were estimated in a Bayesian framework using the R package MCMCglmm. To compare our results with the heritability of morphological traits reported in other avian studies, we also used the ‘prefitModel’ function in the R package RepeatABEL. While RepeatABEL is mainly known for allowing the inclusion of repeated measures, it was chosen as it allows the inclusion of multiple random effects into the model. The package BayesR was used to infer the phenotypic variation explained by SNPs in different effect size distributions and to estimate the SNP-based heritability.

#### Chromosome partitioning of genetic variance
For each chromosome, we estimated the contribution to overall trait variance by fitting a mixed model in MCMCglmm with a GRM estimated from markers on the focal chromosome. We also used GCTA (V1.24) to estimate the variance explained by each chromosome. GRMs were generated with the --make-grm option for each individual chromosome and all but the focal chromosome. Because the model failed to converge when fitting all autosomes simultaneously, we fitted the GRM pairs (e.g. GRM for Chr 1 and GRM for all but Chr 1, using the --mgrm option) in the default REML model one by one to estimate the additive genetic variance explained by each chromosome. For both methods, we used the cor.test function in R to test the significance of a positive Pearson's correlation between chromosome size (based on the zebra finch genome) and variance explained.

#### Genome-wide association scan
To test for association between trait and genotype, we used the RepeatABEL package. The ‘prefitModel’ function first fits a model without the SNP effects and computes the variance components for the trait, using the hglm package. SNPs are then tested for association by accounting for these random effects in the ‘rGLS’ function in RepeatABEL, using the same fixed and random effects as in the heritability analysis. p-values for each SNP are computed using the Wald statistics, together with the SNP effect for each marker. We used a Bonferroni correction to control for family-wise error rate from multiple testing (significance value of 0.05 divided by the number of markers, giving a critical p-value of 1.26 × 10−6).
