
## Genomic data of different resolutions reveal consistent inbreeding estimates but contrasting homozygosity landscapes for the threatened Aotearoa New Zealand hihi

We performed an in-depth inbreeding analysis for hihi (*Notiomystis cincta*), an endemic and nationally vulnerable passerine bird of Aotearoa New Zealand. We first focused on subsampling variants from a reference genome male and then extended our analysis to 10 hihi where low-coverage whole genome resequencing, RAD-seq and SNP array genotypes are available. Detailed methods can be found in [publication](https://doi.org/10.1111/mec.16068).

#### Genomic data and SNP genotyping
Of the 58,466 SNPs included on the array, 45,553 markers passed initial default quality control metrics in the Axiom Analysis Suite software and were classified as “PolyHighResolution” and were used in these analyses. As recommended in Meyermans et al. (2020), no pruning for minor allele frequency or linkage disequilibrium was performed on any of the medium-density genotype data sets (RAD-seq, SNP array data). To explore data sets of different marker density and quality, the loe coverage whole-genome resequencing data for all 10 hihi was divided into marker sets the minimum mean depth (--min-meanDP) filtering threshold value in vcftools. 

#### SNP downsampling for the reference individual
Genotype calls for Yellow were obtained by mapping the long reads back to the reference assembly using Minimap2 v2.17 (Li, 2018) and calling sequence variants using longshot v0.4.1 (Edge & Bansal, 2019). To acquire additional data sets of different SNP densities, the combined data set of 1,593,073 SNPs was then randomly downsampled to subsets of SNPs representing 1/2 (i.e., 796,537 SNPs), 1/4, 1/8, 1/16, 1/32, 1/64, 1/128 and 1/256 (i.e., 6,223 SNPs) of the total, with 10 replicates for each. In addition, 10 replicate data sets sampling the number of SNP array and RAD-seq SNPs at random from the combined data set were also generated.

#### Runs of homozygosity and genome coverage
ROH were first examined with the sliding windows approach in PLINK using the --homozyg function. We calculated the genome coverage parameter (the maximal ROH length the analysis can discover) of our SNP data sets by converting all SNPs in the different data sets to homozygous, determining the total inferred size of homozygous regions >300 kb, and dividing this number by the total assembled genome size of 1,046.4 Mb (Meyermans et al., 2020).

#### Inbreeding coefficients and multilocus heterozygosity
Individual inbreeding coefficients based on ROH. In addition, for the 10 individuals for the lcWGS, SNP array and RAD-seq data sets, genomic inbreeding coefficients FG were calculated using PLINK. PLINK implements the --ibc function from the software GCTA to report three measures of FG, we here report method III, which is based on the correlation between uniting gametes (FIBC). PLINK’s --het function was also used to calculate FG based on observed autosomal homozygous genotype counts for each individual (FHET). MLH was inferred with the lcWGS8 data set using the InbreedR package in R. Correlation coefficients between MLH and all genomic inbreeding estimates for the 10 hihi were calculated using Pearson statistics in R. In addition, we scanned the data sets for the 10 hihi for homozygous-by-descent segments using the R package RZooRoH.
